MapboxSearch/sdk/src/main/java/com/mapbox/search/SearchCallback.kt

```diff
@@ -20,3 +21,24 @@ public interface SearchCallback {
      */
     public fun onError(e: Exception)
 }
+
+/**
+ * Search callback for when only one response is expected
+ */
+public interface SearchResultCallback : SearchCallback {
```

Comment:
I'm not sure if it's worth extending SearchResultCallback from SearchCallback, as users still have the opportunity to override `onResults` and inadvertently disrupt the logic.

Additionally, I'm not sure if it's worthwhile to introduce this callback at all. I don't believe it's a significant issue for customers to handle a single result wrapped in a list. On the other hand, an API that accepts SearchCallback appears more robust and capable of handling various scenarios, such as instances where the backend returns multiple results from different data providers.

Reply:
@DzmitryFomchyn i guess i disagree because it breaks or changes the contract of the method. Now were saying, this method could return multiple results even though your are looking up something explicitly by ID.

Reply:
Got it, makes sense! However, I feel like this interface doesn't have much in common with `SearchCallback`, so I wouldn't inherit it.